cpp/unreal_plugins/UrdfBot/Source/UrdfBot/UrdfBotPawn.cpp

```diff
@@ -0,0 +1,72 @@
+//
+// Copyright(c) 2022 Intel. Licensed under the MIT License <http://opensource.org/licenses/MIT>.
+//
+
+#include "UrdfBotPawn.h"
+
+#include <Camera/CameraComponent.h>
+#include <Components/InputComponent.h>
+
+#include "CoreUtils/Config.h"
+#include "UrdfRobotComponent.h"
+
+AUrdfBotPawn::AUrdfBotPawn(const FObjectInitializer& object_initializer): APawn(object_initializer)
+{
+    // setup robot
+    robot_component_ = CreateDefaultSubobject<UUrdfRobotComponent>(TEXT("RobotComponent"));
+    RootComponent = Cast<USceneComponent>(robot_component_->root_link_component_);
+
+    // setup camera
+    FVector camera_location(
+        Config::get<float>("URDFBOT.URDFBOT_PAWN.CAMERA_COMPONENT.POSITION_X"),
+        Config::get<float>("URDFBOT.URDFBOT_PAWN.CAMERA_COMPONENT.POSITION_Y"),
+        Config::get<float>("URDFBOT.URDFBOT_PAWN.CAMERA_COMPONENT.POSITION_Z"));
+
+    FRotator camera_orientation(
+        Config::get<float>("URDFBOT.URDFBOT_PAWN.CAMERA_COMPONENT.PITCH"),
+        Config::get<float>("URDFBOT.URDFBOT_PAWN.CAMERA_COMPONENT.YAW"),
+        Config::get<float>("URDFBOT.URDFBOT_PAWN.CAMERA_COMPONENT.ROLL"));
+
+    camera_component_ = CreateDefaultSubobject<UCameraComponent>(TEXT("AOpenBotPawn::camera_component_"));
+    ASSERT(camera_component_);
+
+    camera_component_->SetRelativeLocationAndRotation(camera_location, camera_orientation);
+    camera_component_->SetupAttachment(RootComponent);
+    camera_component_->bUsePawnControlRotation = false;
+    camera_component_->FieldOfView = Config::get<float>("URDFBOT.URDFBOT_PAWN.CAMERA_COMPONENT.FOV");
+
+    // debug only
+    robot_component_->test(this);
+    this->AddInstanceComponent(camera_component_);
+}
+
+void AUrdfBotPawn::SetupPlayerInputComponent(class UInputComponent* input_component)
+{
+    Super::SetupPlayerInputComponent(input_component);
+
+    // debug only
+    input_component->BindKey(EKeys::SpaceBar, IE_Pressed, this, &AUrdfBotPawn::test);
+    input_component->BindKey(EKeys::One, IE_Pressed, this, &AUrdfBotPawn::test1);
+    input_component->BindKey(EKeys::Two, IE_Pressed, this, &AUrdfBotPawn::test2);
+}
+
+void AUrdfBotPawn::Tick(float delta_time)
```

Comment:
Do we expect to need this Tick function for anything? If not, we can remove.

Reply:
I believe we will need it for the control API.

Reply:
I'm not sure we will need that. In the case of OpenBot, we need it because OpenBotPawn has an unusual control interface. But we can leave it for now.